cmake/modules/DispatchCompilerWarnings.cmake

```diff
@@ -0,0 +1,85 @@
+
+macro(dispatch_common_warnings)
```

Comment:
This definitely is problematic.  You break cross-compiling for Windows this way.  You should check if the compiler flag is supported and if you are using a GNU frontend before adding these options.  This is also a very large set of flags that you are adding, maintaining this seems like a headache.  How were these handled in autotools?